[Jiang Zhengtao] iP

zhengtaoJ opened the pull request:
# Duke
>*"Life can be ~~easier~~ MUCH easier with a good task manager." -- me*

Duke frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~fast~~SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/AinsleyJ/ip/releases/download/v0.1/ip.jar) *this doesn't work for me though*.
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :smirk:.

And it is **FREE**!

Features:

- [X] Managing tasks
- [X] Managing deadlines, todos, and events
- [ ] Reminders (coming soon) 

Here is a `helloWorld` code just to satisfy the requirement 😝:

```Java
public class Hello {

    public static void main(String[] args){
        System.out.println("Hello, world!");
    }
}
```

Commit:
Add Gradle support

zhengtaoJ committed:
Level-1 features added

zhengtaoJ committed:
Changed the names of classes to avoid confusion.

zhengtaoJ committed:
Changed names of files.

zhengtaoJ committed:
Add and list added.

zhengtaoJ committed:
Add features: mark as done, and unmark.

zhengtaoJ committed:
Todos, events, deadlines added.

zhengtaoJ committed:
Test files added.

zhengtaoJ committed:
Test files added.

zhengtaoJ committed:
Error handling added.

zhengtaoJ committed:
Add command: delete.

zhengtaoJ committed:
Fix format according to the coding standard

zhengtaoJ committed:
Enable saving tasks into hard disk

zhengtaoJ committed:
Use LocalDate for date of deadline

zhengtaoJ committed:
Merge branch 'branch-Level-7'

zhengtaoJ committed:
Resolve merge conflict on level-8

zhengtaoJ committed:
Split InstructionHandler into Ui and Storage

zhengtaoJ committed:
Add packages

zhengtaoJ committed:
Restructure packages

zhengtaoJ committed:
Add Junit tests

zhengtaoJ committed:
Resolve merge conflict on level 8

zhengtaoJ committed:
Add test file

zhengtaoJ committed:
Merge branch 'branch-Level-8'

zhengtaoJ committed:
Genereate jar file

zhengtaoJ committed:
Add JavaDocs

zhengtaoJ committed:
Clean up the codes

zhengtaoJ committed:
Remove test file

zhengtaoJ committed:
Merge branch 'branch-A-JavaDoc'

zhengtaoJ committed:
Merge branch 'branch-A-CodingStandard'

zhengtaoJ committed:
Finish up Level 9

zhengtaoJ committed:
Merge branch 'branch-Level-9'

Btaykb commented on src/main/java/duke/main/Duke.java:
```diff
@@ -0,0 +1,72 @@
+        Ui ui = new Ui(System.out, System.in);
+
+        Duke duke = new Duke(storage, taskManager, ui);
+        duke.run();
```
Good work on the extraction of all the classes

Btaykb commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,185 @@
+
+        switch (type) {
+
+            case "todo":
```
Minor issue, but the case statements do not need an indentation

Btaykb commented on src/main/java/duke/instruction/UnmarkAsDone.java:
```diff
@@ -0,0 +1,80 @@
+     * @param ui The UI to be used by this instruction.
+     */
+    @Override
+    public void act(Ui ui) {
```
Perhaps name of method could be a bit more descriptive

Btaykb reviewed (commented):
Overall good job

sibinhho99 commented on src/main/java/duke/instruction/Add.java:
```diff
@@ -0,0 +1,35 @@
+    }
+
+    /**
+     * Adds the task into the task manager, and return the message when the action is done.
```
The method is singular so it should be `returns` instead of `return`

sibinhho99 commented on src/main/java/duke/instruction/Instruction.java:
```diff
@@ -0,0 +1,112 @@
+    }
+
+    /**
+     * Factory method. Create an instance of subclass of instructions according to the string inputted, and returns it.
```
`Creates` instead of `Create`

sibinhho99 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,185 @@
+import java.util.HashMap;
+
+/**
+ * Represent a task that is stored by main.Duke.
```
`Represents` instead of `Represent`

sibinhho99 commented on src/main/java/duke/parse/Parser.java:
```diff
@@ -0,0 +1,41 @@
+     * @return The instruction generated.
+     * @throws DukeException If the instruction is not recognized.
+     */
+    public Instruction parseInstruction(TaskManager tasks)
```
Perhaps you could move the string parsing logic inside this `Parser` class

zhengtaoJ committed:
Merge remote-tracking branch 'origin/add-gradle-support'

zhengtaoJ committed:
Merged gradle support branch

zhengtaoJ committed:
Add Gradle support

zhengtaoJ committed:
Add checkstyle and fix style

zhengtaoJ committed:
Add Level-10 basic features

zhengtaoJ committed:
Integrate a very ugly UI

zhengtaoJ committed:
MainWindow.fxml: fix file name

Initially there was a typo in the name of the file, which causes the
Main class unable to load the file and throw errors in the jar file.

Change the name from "MainWIndow.fxml" to "MainWindow.fxml".

Otherwise, getResource() is not able to figure out the file and cannot
load it properly.

zhengtaoJ committed:
Add assertions: assert correct filepath

The operations on duke.storage.Storage.java relies on the correct file
path such that it can read/write on the correct data file. Similarly,
duke.main.Main.java relies on the correct fxml file for it to load the
GUI correctly.

Let's add assertions to these two classes. In duke.main.Main.java, add
an assertion to the start() method, such that it ensures that the
filePath is correctly loaded. In duke.storage.storage.readTasks, add an
assertion to ensure that the data file is correctly loaded.

Using assertions ensure that the GUI can be loaded correctly and user
will not loss his/her data.

zhengtaoJ committed:
Add assertion to Storage#writeTasks

zhengtaoJ committed:
Improve code quality

zhengtaoJ committed:
Merge pull request #1 from AinsleyJ/branch-A-CodeQuality

Improve code quality

zhengtaoJ committed:
Merge branch 'A-Assertions'

zhengtaoJ committed:
Add help feature

zhengtaoJ committed:
Fix header comments

zhengtaoJ committed:
Merge branch 'branch-A-Help'

zhengtaoJ committed:
Improve GUI: the dialog box is dynamic now

zhengtaoJ committed:
Add prompt message in GUI

zhengtaoJ committed:
Set title of window

zhengtaoJ committed:
Set theme jekyll-theme-cayman

zhengtaoJ committed:
Add product screenshot

zhengtaoJ committed:
Merge branch 'master' of https://github.com/AinsleyJ/ip

zhengtaoJ committed:
Update user guide

zhengtaoJ committed:
Fix bug in user guide

zhengtaoJ committed:
Change name of screenshot

zhengtaoJ committed:
Update user guide

zhengtaoJ committed:
Fix style

zhengtaoJ committed:
Update link in README.md

zhengtaoJ committed:
Fix user guide
